tests/rbd/test_rbd_immutable_cache_cluster_operations.py

```diff
@@ -0,0 +1,273 @@
+"""Perform cluster-related operations along with
+    immutable object cache test parallel.
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common and fio packages,
+    conf and keyring files
+
+    Test cases covered -
+    CEPH-83574132 - Immutable object cache with cluster operations.
+
+    Test Case Flow -
+
+    1)Create multiple rbd pool and multiple images using rbd commands
+    2)Write some data to the images using FIO
+    3)Create multiple clones images in different pools from the multiple parent image created
+    4)Read the cloned images of different pools in parallel and check the cache status
+    5) Restart the Mon, OSD, and cluster target and parallelly run the IO and cache status
+    6) Remove mon and add mon back to cluster with parallelly run the IO and cache status
+    8) check ceph health status
+    9) Perform test on both Replicated and EC pool
+"""
+import time
+
+from test_rbd_immutable_cache import configure_immutable_cache
+
+from ceph.ceph_admin import CephAdmin
+from ceph.parallel import parallel
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.monitor_workflows import MonitorWorkflows
+from ceph.rbd.initial_config import initial_rbd_config
+from ceph.rbd.utils import getdict, random_string
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.cluster_operations import (
+    check_health,
+    operation,
+    restart_ceph_target,
+    restart_osd,
+)
+from ceph.rbd.workflows.rbd import create_snap_and_clone
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+immutable_cache_path = "/mnt/immutable-cache/"
+
+
+def verify_immutable_cache_objects(rbd, node, **kw):
+    """
+    Method to validate cache objects after configuring immutable cache daemon.
+
+    Args:
+        rbd: RBD object
+        node: cache client node
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+    image_spec = config.get("image_spec")
+    pool = image_spec.split("/")[0]
+    image = image_spec.split("/")[1]
+
+    fio_args = {
+        "client_node": node,
+        "pool_name": pool,
+        "image_name": image,
+        "io_type": "write",
+        "verify": "crc32",
+        "verify_fatal": 1,
+    }
+
+    if kw:
+        fio_args.update(kw)
+
+    try:
+        run_fio(**fio_args)
```

Comment:
don't you think its better to run IOs on mounted path instead of directly on images?

Reply:
No, Immutable cache files are getting created on when user performs IO operation on cloned images so we need to run IO on cloned images. hence running IO's directly on images here.

Reply:
But when would a customer run IOs directly on an image? It's always by mounting right? So how does one use immutable cache in customer scenario?

Reply:
yes but user can use multiple ways of using image to write data mounting is oneway,
even if we mount and run also internally it's rbd image itself
also in test case it didn't mentioned about it may be we can take this further 
but for users ways are more only so in our testing i feel it's good to test with IO on image directly.

Reply:
@manasagowri https://issues.redhat.com/browse/RHCEPHQE-12402
created task to explore customer-based test scenarios to cover more test coverage.